tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -150,20 +188,40 @@ func (a *AwsProvider) CreateWindowsVM() (windowsVM types.WindowsVM, err error) {
 		return nil, fmt.Errorf("failed to get cluster worker IAM, %v", err)
 	}
 
-	// PowerShell script to setup WinRM for Ansible, installing OpenSSH server and open firewall
-	// port number 10250 on the Windows node created
-	userDataWinrm := `<powershell>
-        $url = "https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1"
-        $file = "$env:temp\ConfigureRemotingForAnsible.ps1"
-        (New-Object -TypeName System.Net.WebClient).DownloadFile($url,  $file)
-        & $file
-        Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
-        New-NetFirewallRule -DisplayName "` + types.FirewallRuleName + `"
-        -Direction Inbound -Action Allow -Protocol TCP -LocalPort ` + types.ContainerLogsPort + ` -EdgeTraversalPolicy Allow
+	publicKey, err := a.generatePublicKey()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate public key using private key: %v", err)
+	}
+
+	// PowerShell script to setup WinRM for Ansible, installing OpenSSH server, configuring
+	// public key authentication and open firewall port number 10250 on the Windows node created
+	userData := `<powershell>
+	$url = "https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1"
+	$file = "$env:temp\ConfigureRemotingForAnsible.ps1"
+	(New-Object -TypeName System.Net.WebClient).DownloadFile($url,  $file)
+	& $file
+	Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
```

Comment:
Do we still need to stick to version `0.0.1.0`?

Reply:
I guess we have to as per the Microsoft [doc](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse#installing-openssh-with-powershell).